wandb/sdk/wandb_run.py

```diff
@@ -459,6 +466,21 @@ def _set_iface_port(self, iface_port: int) -> None:
 
     def _telemetry_callback(self, telem_obj: telemetry.TelemetryRecord) -> None:
         self._telemetry_obj.MergeFrom(telem_obj)
+        self._telemetry_obj_updated = True
+        self._telemetry_flush()
+
+    def _telemetry_flush(self) -> None:
+        if not self._telemetry_obj_active:
+            return
+        if not self._telemetry_obj_updated:
+            return
+        if self._backend and self._backend.interface:
+            serialized = self._telemetry_obj.SerializeToString()
+            if serialized == self._telemetry_obj_flushed:
+                return
+            self._backend.interface._publish_telemetry(self._telemetry_obj)
+            self._telemetry_obj_flushed = serialized
+            self._telemetry_obj_updated = False
```

Comment:
if for some reason we fail to update... does it mean that `_telemetry_obj_updated` stays true? should we care about this case?

Reply:
we dont update if the interface isnt set up, in that case we want to keep it marked updated=True so we get around to it later.. this isnt a case that should happen, but we need to check because it is an Optional field.

As for failing,  publish isnt allowed to fail.   if publish generates an exception and is handled by a higher layer, then it is right to keep updated as True so the next time it will get sent.

Reply:
ok makes sense! Thanks!